packages/utils/test/estimate-gas-fee.test.ts

```diff
@@ -0,0 +1,84 @@
+import { estimateGasFees } from '@requestnetwork/utils/src';
+import { BigNumber, providers, Wallet } from 'ethers';
+
+jest.setTimeout(10000);
+
+const mnemonic = 'candy maple cake sugar pudding cream honey rich smooth crumble sweet treat';
+const dummyAddress = '0xf17f52151EbEF6C7334FAD080c5704D77216b732';
+const provider = new providers.JsonRpcProvider('http://localhost:8545');
+const wallet = Wallet.fromMnemonic(mnemonic).connect(provider);
+
+const checkEstimation = (
+  baseEstimation: BigNumber,
+  currentEstimation: BigNumber,
+  ratioMax: number,
+) => {
+  const absRatio =
+    Math.abs(baseEstimation.sub(currentEstimation).toNumber()) /
+    Math.max(baseEstimation.toNumber(), currentEstimation?.toNumber());
+  expect(absRatio).toBeLessThan(ratioMax);
+};
+
+describe('Gas fee estimation', () => {
+  it('Should not be undefined', async () => {
+    const estimation = await estimateGasFees({ provider });
+    expect(estimation.maxFeePerGas).toBeDefined();
+    expect(estimation.maxPriorityFeePerGas).toBeDefined();
+  });
+
+  it('Should return a lower estimation when the previous block is empty', async () => {
+    const firstEstimation = await estimateGasFees({ provider });
+    await provider.send('evm_mine', []);
+    const secondEstimation = await estimateGasFees({ provider });
+
+    expect(
+      firstEstimation.maxFeePerGas?.sub(secondEstimation.maxFeePerGas || 0).toNumber(),
+    ).toBeGreaterThan(0);
+  });
+
+  /**
+   * The EMA used for estimation occur on the range of 100 blocks.
+   * Over 50 blocks we randomly simulate empty or non-empty blocks.
+   * We check that there's no discrepancy between the results:
+   *  - Less than 20% difference between two consecutive estimations
+   *  - Less than 40% difference between the first and the last estimations
```

Comment:
```suggestion
   *  - Less than 40% difference between first and any other estimate
```

Reply:
I think you reviewed this just before I made https://github.com/RequestNetwork/requestNetwork/pull/1072/commits/cc03ae3a26bca3c67e7cdfaef594b16aa074022e 
I think it's better like this:
- No randomization in the test
- The test itself make more sense